### PR TITLE
Downsampling

### PR DESCRIPTION
**Pull request by JoelSaleem**
To run branch locally change values.yaml to:
`airflow.config.AIRFLOW__KUBERNETES__GIT_BRANCH` from `main` to `downsampling`
and 
`dags.git.ref` from `main` to `downsampling`

### CODE REVIEW

File: dags/pipeline.py [hunk 1]
```diff
@@ -188,6 +189,60 @@
     secrets=[aws_secret_id, aws_secret_key]
 )
 
+check_16_bit = KubernetesPodOperator(
```
Comment (udoniyor) [inline dags/pipeline.py:192]: Given majority of the settings are same, can we wrap creation of the KubernetesPodOperator in a function and pass only the attributes that are different between different ones or/and have optional overrides

File: tasks/check_16_bit.py [hunk 2]
```diff
@@ -14,19 +19,38 @@ def run():
...
     makedirs(f'/usr/foo/wav_16bit_{git_hash}')
 
     for wav in listdir('/usr/foo/wavs'):
+        LOGGER.info(f'processing wav: {repr(wav)}')
+        if not wav.endswith('.wav'):
+            continue
+
         num_bits_output = subprocess.run(
-            "ffprobe -v error -show_entries stream=bits_per_sample -of default=noprint_wrappers=1:nokey=1 {wav}",
+            f"ffprobe -v error -show_entries stream=bits_per_sample -of default=noprint_wrappers=1:nokey=1 /usr/foo/wavs/{wav}",
```
Comment (udoniyor) [inline tasks/check_16_bit.py:32]: The directory is hardcoded for foo?

File: tasks/check_16_bit.py [hunk 3]
```diff
@@ -14,19 +19,38 @@ def run():
...
             capture_output=True
         )
-        numbits = num_bits_output.stdout
+        num_bits = num_bits_output.stdout.decode()
+
+        if num_bits_output.stderr:
+            LOGGER.error(num_bits_output.stderr.decode())
+
+        LOGGER.info(f'num_bits {wav}: {num_bits}')
+        if num_bits != '16':  # OR IS IT AN INT?
```
Comment (udoniyor) [inline tasks/check_16_bit.py:42]: did you find out?

File: tasks/check_16_bit.py (hunk 2 above)
Comment (udoniyor) [reply tasks/check_16_bit.py:32]: np: the command could be simplified with a one line bash

`[[ $(ffprobe -v error -show_entries stream=bits_per_sample -of default=noprint_wrappers=1:nokey=1 /usr/foo/wavs/{wav}) == *16* ]] && echo "yay" ||  sox /usr/foo/wavs/{wav} -b 16 {basepath}`

**Review by udoniyor — COMMENTED** [review]
If the `foo` directory is hardcoded on purpose. If the tasks are using the same volume, the task afterwards end up having files from previous tasks?

File: tasks/check_16_bit.py (hunk 2 above)
Comment (JoelSaleem) [reply tasks/check_16_bit.py:32]: foo is the dir (which will be renamed) to which the pv/pvc is mounted. 
When the pipeline is ready, the plan was to generate a unique id or timestamp or input parameter to use as a directory to store the data. Foo (which will be renamed) will still be hardcoded, but the data will be stored as /usr/foo/dag_run_001/output_wavs

For dev simplicity, I have just been leaving it like this for now

File: tasks/check_16_bit.py (hunk 3 above)
Comment (JoelSaleem) [reply tasks/check_16_bit.py]: yes, good catch

**Review by udoniyor — APPROVED** [review]

**Pull request merged**